packages/siopv2-oid4vp-op-auth/src/session/functions.ts

```diff
@@ -130,6 +128,71 @@ export async function createOPBuilder({
   return builder
 }
 
+//fixme: this is written based on OID4VCIHolder.signCallback sync the fixes to this with that function
+export function signCallback(
+  idOpts: ManagedIdentifierOpts,
+  context: IRequiredContext,
+): (jwt: { header: JwtHeader; payload: JwtPayload }, kid?: string) => Promise<string> {
+  return async (jwt: { header: JwtHeader; payload: JwtPayload }, kid?: string) => {
+    const jwk = jwt.header.jwk
+
+    if (!kid) {
+      kid = jwt.header.kid
+    }
+    if (!kid) {
+      kid = idOpts.kid
+    }
+    if (!kid && jwk && 'kid' in jwk) {
+      kid = jwk.kid as string
+    }
+
+    if (kid && !idOpts.kid) {
+      // sync back to id opts
+      idOpts.kid = kid.split('#')[0]
+    }
+
+    const resolution = await context.agent.identifierManagedGet(idOpts)
+    const issuer = jwt.payload.iss || (isManagedIdentifierDidResult(resolution) ? resolution.did : resolution.issuer)
+    if (!issuer) {
+      return Promise.reject(Error(`No issuer could be determined from the JWT ${JSON.stringify(jwt)}`))
+    }
+    if (kid && isManagedIdentifierDidResult(resolution) && !kid.startsWith(resolution.did)) {
+      // Make sure we create a fully qualified kid
+      const hash = kid.startsWith('#') ? '' : '#'
+      kid = `${resolution.did}${hash}${kid}`
+    }
+    const header = { ...jwt.header, ...(kid && !jwk && { kid }) } as Partial<JWTHeader>
+    const payload = { ...jwt.payload, ...(issuer && { iss: issuer }) }
+    if (jwk && header.kid) {
+      delete header.kid
+    }
+
+    const result: JwsCompactResult = await context.agent.jwtCreateJwsCompactSignature({
+      issuer: { identifier: issuer, noIdentifierInHeader: false },
+      protectedHeader: header,
+      payload,
+    })
+    return result.jwt
+  }
+}
+
+function getVerifyJwtCallback(
+  resolver?: Resolvable,
+  verifyOpts?: JWTVerifyOptions & {
+    checkLinkedDomain: 'never' | 'if_present' | 'always'
+    wellknownDIDVerifyCallback?: VerifyCallback
+  },
+): VerifyJwtCallback {
+  return async (jwtVerifier, jwt) => {
+    resolver = resolver ?? getResolver({ subjectSyntaxTypesSupported: ['ethr', 'ion'] })
+    const audience =
+      jwtVerifier.type === 'request-object' || jwtVerifier.type === 'id-token' ? (verifyOpts?.audience ?? getAudience(jwt.raw)) : undefined
+    //todo probably wise to revisit this. this is called verifyDidJWT and expects a did resolver param.
+    await verifyDidJWT(jwt.raw, resolver, { audience, ...verifyOpts })
```

Comment:
Why is this calling verifyDidJwt instead of the new signature service?

Reply:
discussed this, for now returning true and added a todo with the ticket number for refactoring it